lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -172,45 +184,48 @@ define((require, exports, module) ->
       ]
       LABEL_PARAMETER_INDEX: 2
 
-      getWholeEquation: (range) ->
+      getMacrosArgumentRange: (session, argumentStartPos) ->
+        argumentRange = findSurroundingBrackets(session, argumentStartPos)
+        if argumentRange.mismatch
+          return null
+        else
+          return new Range(argumentRange.start.row, argumentRange.start.column + 1,
+                           argumentRange.end.row, argumentRange.end.column)
+
+      getWholeEquation: (tokenIterator) ->
 
         tokenValues = []
         labelSequenceIndex = 0
         labelParameters = []
         curLabelParameter = null
         curLabelTokens = []
 
-        tokenIterator = new TokenIterator(editor.getSession(), range.start.row, range.start.column)
-        tokenIterator.stepForward() # at first it should be on the closing token of begin sequence
-        {row: curRow, column: curColumn} = tokenIterator.getCurrentTokenPosition()
+        session = editor.getSession()
+        range = tokenIterator.range
         token = tokenIterator.getCurrentToken()
-
-        while range.contains(curRow, curColumn + token.value.length)
-          tokenToMatch = ch.LABEL_SEQUENCE[labelSequenceIndex]
-
-          if token.type == tokenToMatch.type and tokenToMatch.value.test(token.value)
-
-            if labelSequenceIndex == ch.LABEL_PARAMETER_INDEX
-              curLabelParameter = token.value
-
-            curLabelTokens.push(token)
-            labelSequenceIndex += 1
-
-            if labelSequenceIndex == ch.LABEL_SEQUENCE.length
-              labelParameters.push(curLabelParameter)
-              labelSequenceIndex = 0
-              curLabelTokens = []
-
-          else
-            labelSequenceIndex = 0
-            for labelToken in curLabelTokens
-              tokenValues.push(labelToken.value)
-            curLabelTokens = []
+        tokenPosition = tokenIterator.getCurrentTokenPosition()
+
+        while token?
+          if not token?
+            break
+
+          noPush = false
```

Comment:
pls rename to `acceptToken = true`. Double negation `if not noPush` blows the mind